File: 백준/Bronze/17608. 막대기/막대기.swift
import Foundation

var sticks = [Int]()
for _ in 0..<Int(readLine()!)! {
    sticks.append(Int(readLine()!)!)
}
var stack = [Int]()
var maxHeight = 0
for stick in sticks.reversed() where stick > maxHeight {
    stack.append(stick)
    maxHeight = stick
}
print(stack.count)

File: 백준/Bronze/17608. 막대기/막대기.py
import sys

input = sys.stdin.readline
n = int(input())
sticks = [int(input()) for _ in range(n)]
visible_count = 0
max_height = 0
for stick in reversed(sticks):
    if stick > max_height:
        visible_count += 1
        max_height = stick
sys.stdout.write(str(visible_count))